client/src/reducer.js

```diff
@@ -3,6 +3,13 @@ export default function reducer (state, action) {
 	case 'EXAMPLE_MUTATION':
 		state.example = action.payload;
 		return state;
+		
+	case 'BUY_GENERATOR':
+		store.state = action.payload;
+		state.counter = expected.counter;
+		state.generators = expected.generators;
```

Comment:
should loop through a list of generators and find which generator that user is buying